### PR TITLE
Standardize/Simplify code style and add STOP functionality

### PR DESCRIPTION
**Pull request by nemchik**
I found lots of places where the code has non standard spacing such as if () rather than if()
I also removed the use of variables being declared for each button that needs to be pressed (there is no need to store this in memory).
I also switched everything to querySelector rather than getElementById or class.

These commits make it fairly easy to open almost any file and use it as a template to add a new music payer with very minimal code. This also makes updating the keysocket code to accomodate player changes a little bit simpler.

Another point of note (forgot previously) is I made sure the if statements for which key is pressed are in alphabetical order and in the same order for each socket (ex next, play, prev, stop)

### CODE REVIEW

File: extension/keysocket-amazon-cloud-player.js [hunk 1]
```diff
@@ -1,12 +1,9 @@
...
-        simulateClick(nextButton);
-    } else if(key === PLAY) {
-        var playPauseButton = document.querySelector('[playeraction=togglePlay]');
-        simulateClick(playPauseButton);
-    } else if(key === PREV) {
-        var backButton = document.querySelector('[playeraction=previous]');
-        simulateClick(backButton);
-    }
+	if(key === NEXT) {
+		simulateClick(document.querySelector('[playeraction=next]'));
```
Comment (borismus) [inline extension/keysocket-amazon-cloud-player.js]: Please stick to two space indentation throughout.

File: extension/keysocket-deezer.js [hunk 2]
```diff
@@ -1,38 +1,23 @@
...
 		} else if(key === PLAY) {
-			var isPlaying = document.getElementById('player_control_play').style.display === 'none';
-			var playPauseButton = null;
-			if(isPlaying) {
-			  playPauseButton = document.getElementById('player_control_pause');
-			} else {
-			  playPauseButton = document.getElementById('player_control_play');
-			}
-			simulateClick(playPauseButton);
+			simulateClick(document.querySelector('#player_control_play').style.display === 'none' ? document.querySelector('#player_control_pause') : document.querySelector('#player_control_play'));
```
Comment (borismus) [inline extension/keysocket-deezer.js]: Please do not put the whole block in one line.

File: extension/keysocket-bop.js [hunk 3]
```diff
@@ -1,12 +1,9 @@
...
-    } else if(key === PREV) {
-        var backButton = document.getElementsByClassName('icon-previous')[0];
-        simulateClick(backButton);
-    }
+	if(key === NEXT) {
+		simulateClick(document.querySelector('.icon-next')[0]);
+	} else if(key === PLAY) {
+		simulateClick(document.querySelector('.wrapper-icon-play-pause')[0]);
+	} else if(key === PREV) {
+		simulateClick(document.querySelector('.icon-previous')[0]);
```
Comment (borismus) [inline extension/keysocket-bop.js]: There is really no need to combine these two into one line.

File: extension/keysocket-hypem.js [hunk 4]
```diff
@@ -1,9 +1,9 @@
 function onKeyPress(key) {
-    if(key === NEXT) {
-        simulateClick(document.querySelector('#playerNext'));
-    } else if(key === PLAY) {
-        simulateClick(document.querySelector('#playerPlay'));
-    } else if(key === PREV) {
-        simulateClick(document.querySelector('#playerPrev'));
-    }
+	if(key === NEXT) {
```
Comment (borismus) [inline extension/keysocket-hypem.js]: Are these tabs?

File: extension/keysocket-plex.js [hunk 5]
```diff
@@ -1,26 +1,11 @@
...
-        var backButton = document.querySelector('button.previous-btn');
-        simulateClick(backButton);
-    } else if(key === STOP) {
-        var stopButton = document.querySelector('button.stop-btn');
-        simulateClick(stopButton);
-    }
+	if(key === NEXT) {
+		simulateClick(document.querySelector('button.next-btn'));
+	} else if(key === PLAY) {
+		simulateClick(document.querySelector('.play-btn').classList.contains('hidden') ? simulateClick(document.querySelector('button.pause-btn')) : simulateClick(document.querySelector('.play-btn')));
```
Comment (borismus) [inline extension/keysocket-plex.js]: Line is way too long. Please constrain to 80 chars.

**Comment by borismus** [comment]
Thanks. After you address my comments, could you also thoroughly test to make sure that none of the players are broken? Thanks.

File: extension/keysocket-hypem.js (hunk 4 above)
Comment (nemchik) [inline extension/keysocket-hypem.js]: Yeah tabs, my editor converted, sorry I will fix them.

To confirm the preferred indentation is 4 spaces, yes?

Comment (borismus) [inline extension/keysocket-hypem.js]: 2 spaces throughout please.

File: extension/keysocket-deezer.js (hunk 2 above)
Comment (nemchik) [inline extension/keysocket-deezer.js]: Using inline if prevents storing variables, would this be an acceptable alternative?
```
      simulateClick(
        document.querySelector('#player_control_play').style.display === 'none'
      ?
        document.querySelector('#player_control_pause')
      :
        document.querySelector('#player_control_play')
      );
```

Comment (borismus) [inline extension/keysocket-deezer.js]: This is way too clever and premature optimization. 

An extra variable will go a long way for readability. Under the hood these local variables don't amount to much, they are probably completely optimized out.

Comment (nemchik) [inline extension/keysocket-deezer.js]: how about something like this?
```
function onKeyPress(key) {
  var clickButton;
  if(document.querySelector('#player_control_play')) { // old deezer style
    if(key === NEXT) {
      clickButton = document.querySelector('#player_control_next');
    } else if(key === PLAY) {
      if(document.querySelector('#player_control_play').style.display === 'none') {
        clickButton = document.querySelector('#player_control_pause');
      } else {
        clickButton = document.querySelector('#player_control_pause');
      }
    } else if(key === PREV) {
      clickButton = document.querySelector('#player_control_prev');
    } else if(key === STOP) {
      clickButton = document.querySelector('#player_control_pause');
    }
  } else { // new deezer style
    if(key === NEXT) {
      clickButton = document.querySelector('.control-next')[0);
    } else if(key === PLAY) {
      if(document.querySelector('.control-play')[0]) {
        clickButton = document.querySelector('.control-play')[0];
      } else {
        clickButton = document.querySelector('.control-pause')[0];
      }
    } else if(key === PREV) {
      clickButton = document.querySelector('.control-prev')[0];
    } else if(key === STOP) {
      clickButton = document.querySelector('.control-pause')[0];
    }
  }
  simulateClick(clickButton);
}
```

Comment (borismus) [inline extension/keysocket-deezer.js]: Whoa. Please separate into functions. Also you might benefit from a
switch() statement.

Comment (nemchik) [inline extension/keysocket-deezer.js]: I don't understand what you mean. This code is very similar to the original code, with the exception that the simulateClick is called at the end.

Can you provide an example of what you're looking for and i'll try and get the rest of the sockets to match?

Comment (borismus) [inline extension/keysocket-deezer.js]: Make more variables.

var isNewUI = ...code to detect whether or not it's the new UI...;
if (isNewUI) {
  handleNewDeezerUI()
} else {
  handleOldDeezerUI()
}

Also, in the interest of simplifying the code, how about a helper like simulateClickOnSelector('#mySelector').

Comment (nemchik) [inline extension/keysocket-deezer.js]: I think I see the direction you're wanting to go. This will make the code a lot bigger in terms of what you're looking at visually when you open a file. I don't really think this will simplify the code.

As for the case switch, I like the idea, but I don't think it'll be helpful in the goal of simplifying things. I don't think case switch will work any better than if/else for the majority of the code here. 

I'll write up a mock to try out the example you posted and I'll put it here in the comments.

Another file I have concerns with is the YouTube socket. The changes I posted work, but looking at the code isn't obvious as to what's going on. I'll probably have to rewrite that one.

**Comment by borismus** [comment]
Please rebase.

**Comment by nemchik** [comment]
What is rebase?

**Comment by feedbee** [comment]
It's not a good practice to place not-related changes in one PR. It's harder to track the nature of every change. And it's not possible to merge one type of changes without merging another the change of another type.

Currently, after PR #100 was merged, there will be conflicts in all of the files, in almost every line of code. The only way to rebase it is override all my changes.

What standards did you used to format the code? Why space after `if` was removed? I looked through a couple of standards found in Google and none of them promoted `if() {`, all of them suggested to use `if () {`. Some links: [Google JavaScript Style Guide](http://google-styleguide.googlecode.com/svn/trunk/javascriptguide.xml), [Douglas Crockford’s JavaScript coding style](http://javascript.crockford.com/code.html), [more](https://github.com/Seravo/js-winning-style).

And did you used any tool to check you result code automatically?

**Comment by nemchik** [comment]
I havent used anything to automatically check code. I had thought if was treated as a function and as such should not need the space after, if this is incorrect then i don't at all need this merged.

I was trying to accomplish the same thing in this pull as it seems you have accomplished in #100 . An additional intention was to include the stop functionality which is not at all complex or large to include, but i'm not amazing with git so i don't think i'll get this accomplished unless I do one player a

**Comment by feedbee** [comment]
OK, now you have to choose one of: rebase/merge you work (which is hard to do other way than to override #100), or to start from the beginning. Git is good tool, but it requires good understanding of it. Otherwise it will drive you mad.

What if to start from the beginning? By the way, it's better to create Pull Requests from branches in forked repository (not from master). So create new branch like `stop-functionality`, make changes which does nothing other than stop functionality. Test it. Commit it in one commit (less commits is better), push create PR. Let this PR be merged.

After, if you want to do refactoring (like removing redundant variables), do it in another branch and create another PR.

May be all of this is not my business, but the reasons I write it are to get safe the work I did in PR #100, and to help you contribute new stop functionality.

**Comment by nemchik** [comment]
I actually really appreciate the advice as I know nothing about git. This
is my first time actually using git and as you can see I've probably made
every rookie mistake available.

My intention isn't to anger anyone or step on anyone else's toes with their
work, but like you said, I want to contribute, and there are features
(stop) that I'd like to see in this extension.

**Comment by feedbee** [comment]
So, what about to recreate your fork from scratch (as you already did earlier, as I can see), make new branch, move the changes about stop functionality into fresh code, test, commit and make new clean PR?

**Comment by nemchik** [comment]
That works for me.

And in thinning this through I would like to start with a different feature
that Boris mentioned in a comment, so I'll make that and wait to see if it
merges and then work on stop.

Thank you for your input. Is there a way I may contact you for advice or
other questions (if you don't mind that is; I'll try not to pester you much)

**Comment by feedbee** [comment]
My email is available at my git account page.

**Comment by nemchik** [comment]
Thank you again.
